src/control/control.c

```diff
@@ -305,10 +296,9 @@ void dt_control_shutdown(dt_control_t *s)
   if(!s)
     return;
   dt_pthread_mutex_lock(&s->cond_mutex);
-  dt_pthread_mutex_lock(&s->run_mutex);
-  const gboolean was_running = s->running;
-  s->running = FALSE;
-  dt_pthread_mutex_unlock(&s->run_mutex);
+
+  gint was_running = 0;
```

Comment:
Remove "0;" (i.e. set to return of next call) unless the change in behavior is intentional.

Reply:
Yes i also spotted that already and indeed the call was just wrong.